tests/integration/conftest.py

```diff
@@ -72,99 +81,101 @@ def fixture_test_db_flask_image(pytestconfig: Config):
-        None,
-    )
-
-    if not charm_file:
-        tmp_charm_location = tmp_path_factory.mktemp(framework)
-        charm_location = PROJECT_ROOT / f"examples/{framework}/charm"
-        if framework == "flask":
-            charm_location = PROJECT_ROOT / f"examples/{framework}"
+@pytest.fixture(scope="module", name="test_db_flask_image")
+def fixture_test_db_flask_image(pytestconfig: Config):
+    """Return the --test-flask-image test parameter."""
+    test_flask_image = pytestconfig.getoption("--test-db-flask-image")
+    if not test_flask_image:
+        raise ValueError("the following arguments are required: --test-db-flask-image")
+    return test_flask_image
 
-        charmcraft_yaml = yaml.safe_load((charm_location / "charmcraft.yaml").read_text())
-        charmcraft_yaml.update(charm_dict)
-        (tmp_charm_location / "charmcraft.yaml").write_text(yaml.dump(charmcraft_yaml))
-        charm_file = await ops_test.build_charm(charm_location)
-    elif charm_file[0] != "/":
-        charm_file = PROJECT_ROOT / charm_file
-    inject_venv(charm_file, PROJECT_ROOT / "src" / "paas_charm")
-    return pathlib.Path(charm_file).absolute()
 
+@pytest.fixture(scope="module", name="expressjs_app_image")
+def fixture_expressjs_app_image(pytestconfig: Config):
+    """Return the --expressjs-app-image test parameter."""
+    image = pytestconfig.getoption("--expressjs-app-image")
+    if not image:
+        raise ValueError("the following arguments are required: --expressjs-app-image")
+    return image
 
-async def build_charm_file(
-    pytestconfig: pytest.Config, ops_test: OpsTest, tmp_path_factory, framework
-) -> str:
-    """Get the existing charm file if exists, build a new one if not."""
-    charm_file = next(
-        (f for f in pytestconfig.getoption("--charm-file") if f"/{framework}-k8s" in f), None
-    )
 
-    if not charm_file:
-        charm_location = PROJECT_ROOT / f"examples/{framework}/charm"
-        if framework == "flask":
-            charm_location = PROJECT_ROOT / f"examples/{framework}"
-        charm_file = await ops_test.build_charm(charm_location)
-    elif charm_file[0] != "/":
-        charm_file = PROJECT_ROOT / charm_file
-    inject_venv(charm_file, PROJECT_ROOT / "src" / "paas_charm")
-    return pathlib.Path(charm_file).absolute()
+@pytest.fixture(scope="module", name="flask_minimal_app_image")
+def fixture_flask_minimal_app_image(pytestconfig: Config):
+    """Return the --expressjs-app-image test parameter."""
+    image = pytestconfig.getoption("--flask-minimal-app-image")
+    if not image:
+        raise ValueError("the following arguments are required: --flask-minimal-app-image")
+    return image
 
 
-async def build_charm_file_with_config_options(
+def build_charm_file(
     pytestconfig: pytest.Config,
-    ops_test: OpsTest,
-    tmp_path_factory,
     framework: str,
-    config_options: dict,
+    tmp_path_factory,
+    charm_dict: dict = {},
```

Comment:
Objects should not be used as default argument values in Python because things are pass by reference.
For example,

```
def default_test(d={}):
    print(d)
    d["a"] = "b"

default_test() # prints {}
default_test() # prints {"a":"b"}
```

Reply:
The best practice is to default it to None and initialize it in the body like so:
```
def default_test(d=None):
   if d is None:
       d = {}
```